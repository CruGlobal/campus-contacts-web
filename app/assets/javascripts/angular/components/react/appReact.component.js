import template from './appReact.html';

angular.module('missionhubApp').component('appReact', {
    controller: AppReactController,
    template: template,
});

function AppReactController($rootScope) {}
