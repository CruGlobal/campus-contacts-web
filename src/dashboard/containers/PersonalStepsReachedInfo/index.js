import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import { useQuery } from 'react-apollo-hooks';
import gql from 'graphql-tag';

import AppContext from '../../appContext';
import Header from '../../components/Header';

const GET_IMPACT_REPORT = gql`
    query impactReport($organizationId: ID!) {
        impactReport(organizationId: $organizationId) {
            pathwayMovedCount
        }
    }
`;

const PersonalStepsReachedInfo = () => {
    const { orgId } = useContext(AppContext);

    const { data, loading } = useQuery(GET_IMPACT_REPORT, {
        variables: {
            organizationId: orgId,
        },
    });
    const { t } = useTranslation('insights');

    if (loading) {
        return <Header>{t('loading')}</Header>;
    }

    const { impactReport: report } = data;

    return (
        <Header>
            {t('personalSteps.reached', { count: report.pathwayMovedCount })}
        </Header>
    );
};

export default PersonalStepsReachedInfo;
