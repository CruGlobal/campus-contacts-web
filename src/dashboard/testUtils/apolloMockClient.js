import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
import { buildClientSchema } from 'graphql/utilities';
import { createMockerFromIntrospection } from 'fraql/mock';
import gql from 'fraql';

import introspectionQuery from '../../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument, options) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
