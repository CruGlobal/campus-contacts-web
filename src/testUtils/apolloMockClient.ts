import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
// @ts-ignore
import { buildClientSchema } from 'graphql/utilities';
// @ts-ignore
import { createMockerFromIntrospection } from 'fraql/mock';
// @ts-ignore
import gql from 'fraql';

import introspectionQuery from '../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument: any, options: any) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
