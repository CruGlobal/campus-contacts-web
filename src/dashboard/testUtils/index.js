import React, { ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';

import { createApolloMockClient } from './apolloMockClient';
import defaultTheme from '../defaultTheme';
import { ThemeProvider } from 'emotion-theming';
import AppContext from '../appContext';

export const renderWithContext = (component, { mocks } = {}, context = {}) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={context}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });

    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
