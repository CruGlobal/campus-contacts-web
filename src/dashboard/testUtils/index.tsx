import React, { ReactNode } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';

import defaultTheme from '../defaultTheme';
import AppContext from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

export const renderWithContext = (
    component: any,
    { mocks: mocks = {}, appContext = {} } = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    // @ts-ignore
    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
