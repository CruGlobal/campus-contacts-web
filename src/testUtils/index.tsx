import React, { ReactNode, ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import { IMocks } from 'graphql-tools';

import defaultTheme from '../defaultTheme';
import {
    AppContext,
    AppContextValue,
    appContextDefaultValue,
} from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

interface RenderWithContextParams {
    mocks?: IMocks;
    appContext?: AppContextValue;
}

export const renderWithContext = (
    component: ReactElement,
    {
        mocks: mocks = {},
        appContext = appContextDefaultValue,
    }: RenderWithContextParams = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children?: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
