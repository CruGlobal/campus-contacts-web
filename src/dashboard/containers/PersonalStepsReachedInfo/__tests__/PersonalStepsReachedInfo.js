// Vendors
import React from 'react';
import { waitForElement } from '@testing-library/react';

import { renderWithContext } from '../../../testUtils';
import PersonalStepsReachedInfo from '../';

describe('<PersonalStepsReachedInfo />', () => {
    it('should render properly loading state', async () => {
        renderWithContext(<PersonalStepsReachedInfo />).snapshot();
    });

    /*
    it('should render properly with data', async () => {
        const { snapshot, getByText } = renderWithContext(
            <PersonalStepsReachedInfo />,
            {
                mocks: {
                    ImpactReport: () => ({
                            pathwayMovedCount: () => 13
                    }),
                    impactReport: () => ({
                        pathwayMovedCount: () => 13
                    }),
                    Query: () => ({
                        ImpactReport: () =>({
                            pathwayMovedCount: () => 13
                        }),
                        impactReport: () => ({
                            pathwayMovedCount: () => 13
                        }),
                    })
                },
            },
        );

        await waitForElement(() => getByText('This year'));
        snapshot();
    });
    */
});
