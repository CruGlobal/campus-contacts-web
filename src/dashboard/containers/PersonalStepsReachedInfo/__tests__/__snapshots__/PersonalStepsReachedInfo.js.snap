// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalStepsReachedInfo /> should render properly loading state 1`] = `
<div>
  <h1
    class="css-1c2npbj"
  >
    Loading...
  </h1>
</div>
`;
